kinto/core/initialization.py

```diff
@@ -413,13 +413,62 @@ def __call__(self, event):
 
 
 class EventResourceIdFilter:
-    pass
+    def __init__(self, resource_ids, config):
+        self.resource_ids = resource_ids
+
+    def phash(self):
+        return 'for_resource_id = {}'.format(','.join(self.resource_ids))
+
+    def __call__(self, event):
+        request = event.request
+        uri = event.payload.get('uri')
+        r_name, matchdict = utils.view_lookup(request, uri)
+        """
+        Special case for Create bucket.
+
+        """
+        if event.payload.get('action') == 'create' and r_name == 'bucket':
+            return True
+        for resource_id in self.resource_ids:
+            returned_resource_ids = utils.parse_resource(resource_id)
+            if event.payload.get('action') == 'create':
+                if r_name == 'collection':
+                    bucket = matchdict.get('bucket_id')
+                    if bucket == returned_resource_ids.get('bucket'):
+                        return True
+                elif r_name == 'record':
+                    bucket = matchdict.get('bucket_id')
+                    collection = matchdict.get('collection_id')
```

Comment:
you can put those `matchdict.get()` and `returned_resource_ids.get()` above the `if` to ease reading

Reply:
ok, going out in the next commit